# Week 6 : Case Study 2 - Bimal H M

## Measures of Central Tendency and Dispersion

### STEP 1 : Importing all Necessary Libraries in to Notebook

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### STEP 2 : Reading the Data set in to Notebook 

In [2]:
data = pd.read_csv('StudentsPerformance.csv')

### STEP 3 : Displaying the Dataset 

#### A) Displaying Dataset Info

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


#### B) Displaying First few Rows

In [4]:
data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

### Q1 : Find out how many males and females participated in the test.

In [5]:
data.groupby('gender')['gender'].count()

gender
female    518
male      482
Name: gender, dtype: int64

##### There are 518 females and 482 males participated in the test

### Q2 : What do you think about the students' parental level of education?

In [6]:
data.groupby('parental level of education')['parental level of education'].count()

parental level of education
associate's degree    222
bachelor's degree     118
high school           196
master's degree        59
some college          226
some high school      179
Name: parental level of education, dtype: int64

### Q3 : Who scores the most on average for math, reading and writing ?

#### A)  Based on Gender

In [7]:
data.groupby('gender')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

##### In maths test males scored more marks on average than females, While in other two tests females scored more marks on average 

#### B) based on Test preparation course ?

In [8]:
data.groupby('test preparation course')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

##### In all 3 tests, Those who completed the test prepartion course scored high on average

### Q4 : What do you think about the scoring variation for math, reading and writing?

#### A)  Based on Gender

In [9]:
data.groupby('gender')[['math score','reading score','writing score']].var()

math score  reading score  writing score
gender                                          
female  256.958593     207.677438     220.369327
male    211.889097     200.211010     202.413924

##### In all 3 tests females scores show more variance than male scores

#### B) based on Test preparation course ?

In [10]:
data.groupby('test preparation course')[['math score','reading score','writing score']].var()

math score  reading score  writing score
test preparation course                                          
completed                210.884027     183.265864     175.202612
none                     246.668662     213.419851     226.251739

##### In all 3 tests those who completed test prepartion course show less variance in scores.

### 5) The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.

##### Since we need top 25%, We have to find 3rd Quartile value (Q3) which devide data in to 3:1 ratio, with 1 representing top 25%. To get Q3 We can either use describe function on "Math Score' as follows and get the value corresponds to 75% from there, as follows.

In [11]:
data.describe()['math score']

count    1000.000000
mean       66.001000
std        15.569567
min         0.000000
25%        57.000000
50%        66.000000
75%        77.000000
max       100.000000
Name: math score, dtype: float64

##### Or we can directly find it out by 

In [12]:
data.quantile(0.75)['math score']

77.0

##### From Q3 we can see any student who scored above 77.0 mark in maths exam are eligible for bonus mark. Now lets check for how many students are eligible.

In [13]:
d1 = data['math score']
d1[d1>77.0].count()

231

##### There are 231 students who scored above 77.0 mark in maths who is eligible for bonus mark.